### PR TITLE
nixos/libvirtd: make it possible to boot a UEFI aarch64 vm on x86_64

### PR DESCRIPTION
**Pull request by symphorien**
###### Description of changes

to boot UEFI vms of several architectures you need both a aarch64 efi and x86 uefi implem. Only the native one is currently available.vThis change makes it possible to have several of them. Should I also make it the default?

tested by booting the aarch64 iso of nixos

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/libvirtd.nix [hunk 1]
```diff
@@ -141,9 +147,9 @@ in
     (mkRenamedOptionModule
       [ "virtualisation" "libvirtd" "qemuOvmf" ]
       [ "virtualisation" "libvirtd" "qemu" "ovmf" "enable" ])
-    (mkRenamedOptionModule
+    (mkRemovedOptionModule
       [ "virtualisation" "libvirtd" "qemuOvmfPackage" ]
-      [ "virtualisation" "libvirtd" "qemu" "ovmf" "package" ])
+      "If this option was set to `foo`, set the option `virtualisation.libvirtd.qemu.ovmf.packages' to `[foo.fd]` instead.")
```
Comment (Mindavi) [inline nixos/modules/virtualisation/libvirtd.nix:152]: Not sure about this line/change.

**Comment by Mindavi** [comment]
Diff LGTM, just the one question.

File: nixos/modules/virtualisation/libvirtd.nix (hunk 1 above)
Comment (symphorien) [reply nixos/modules/virtualisation/libvirtd.nix:152]: I don't know what you find unclear, so maybe you can suggest a formulation that would be clearer to you?

The point is that if before you had `virtualisation.libvirtd.qemuOvmfPackage = foo` you should replace that by `virtualisation.libvirtd.qemu.ovmf.packages = [ foo.fd ]`.

Comment (Mindavi) [reply nixos/modules/virtualisation/libvirtd.nix:152]: ah, it will just show that as an error message. That sounds ok to me.

**Comment by Mechtecs** [comment]
Thank you very much! I am trying to do this for a few months now, every time I have a bit of free time.

**Pull request merged**